### DISCUSSION TITLE
handleDelete isn't permanent

### DISCUSSION THREAD

**Discussion by Charlotte-Ketteler** [neuefische/questions#671] [Web] [answered]
I am building a Booking-App where you can book riding lessons. I have a BookingOverview Component where I implemented a delete button. It deletes the booked appointment. The issue is, when leaving the BookingOverviewPage and returning, the booking reappears. How can I delete booking permanently? [Branch: Terminübersicht](https://github.com/Charlotte-Ketteler/capstone-project/tree/Termin%C3%BCbersicht). Thanks for your help! :)

**Comment by Kheirah** [comment, marked as answer]
The issue is that you're not deleting the entry from your bookings at all; you're merely making your JSX markup invisible on a click of a button. I made two comments in the code below to explain in more detail.
```js
import Link from "next/link";
import { HorsesList } from "@/lib/data";
import { ImageCowgirlOnWhiteHorse } from "../../Styles/ImageCowgirlOnWhiteHorse";
import { CardStyled } from "@/Styles/CardStyled";
import { useState } from "react";

export default function BookingOverview({ recentlyBooked, timeSlots }) {
  const [toDelete, setToDelete] = useState(false);

  function handleDelete() {
    setToDelete(true); //You are using a local state in order to control the visibility of `CardStyled`
  }

  return (
    <>
      {/* when toDelete is true, you're showing nothing on the page, but the booking still exists */}
      {toDelete ? null : (
        <CardStyled>
          <ImageCowgirlOnWhiteHorse />
          <p>Uhrzeit: {recentlyBooked?.startTime}</p>
          <div>Personen: {recentlyBooked?.numberOfPeople}</div>
          <p> Pferde:</p>
          <ul>
            {recentlyBooked?.horses.map((horseID) => (
              <li key={horseID}>
                {HorsesList.find((horse) => horse.id === horseID).name}
              </li>
            ))}
          </ul>
          <button onClick={handleDelete}>löschen</button>
        </CardStyled>
          
      )}
      
        <Link href="/">Back to Homepage</Link>
        <br />
        <Link href="/BookingPage">Termin Buchen</Link>
     
    </>
  );
}
```

If you want to delete an entry from your bookings, you need to create a `handleDelete` function in `_app.js`.
In that function, you `filter` through your `timeSlots` (that's your state) and filter out the entry. You need the `id` for doing that.

Try creating that function first, write a console.log in that function and pass the function down such that you see that it works when you click the delete button. Then go about implementing the filter logic. Check the session about deleting entries from a state when you need guidance on how to do that.

**Reply by Charlotte-Ketteler** [reply]
Thank you very much, Ernst! :)
